### PR TITLE
Repair crashed car

### PR DESCRIPTION
**Pull request by Konrad-Rys**
@sebastian-mikucki , @dominik1102 , @tbradlo , @rmycek please review

### CODE REVIEW

File: src/main/java/com/guysfromusa/carsgame/services/CarService.java [hunk 1]
```diff
@@ -117,4 +116,11 @@ public CarEntity addCarToGame(String carName, String gameName, Point startingPoi
         return carRepository.save(car);
     }
 
+    public CarEntity repairCar(String name) {
+        CarEntity car = carRepository.findByName(name)
+                .orElseThrow(() -> new EntityNotFoundException("Car with name " + name + " does not exist"));
+
+        car.setCrashed(false);
+        return carRepository.save(car);
```
Comment (sebastian-mikucki) [inline src/main/java/com/guysfromusa/carsgame/services/CarService.java]: if the entity modification is in transaction then we don't need to perform save on repository

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java [hunk 2]
```diff
@@ -84,4 +82,15 @@ public Long removeCar(@ApiParam(name = "name", value="Car name to be deleted") @
         return carService.deleteCarByName(name);
     }
 
+    @PostMapping(path= "{name}/fix")
+    @ApiOperation(value = "Repair crashed car", response = Car.class)
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Car successfully repaired"),
+        @ApiResponse(code = 400, message = "Car with given name does not exist")
```
Comment (sebastian-mikucki) [inline src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: EntityNotFound is translated to 404

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java [hunk 3]
```diff
@@ -84,4 +82,15 @@ public Long removeCar(@ApiParam(name = "name", value="Car name to be deleted") @
         return carService.deleteCarByName(name);
     }
 
+    @PostMapping(path= "{name}/fix")
```
Comment (sebastian-mikucki) [inline src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: I think better to use PutMapping as finally this is an update of the car

Comment (sebastian-mikucki) [reply src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: and I'm thinking about this /fix as the only think we can update is to make car not crashed

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java [hunk 4]
```diff
@@ -84,4 +82,15 @@ public Long removeCar(@ApiParam(name = "name", value="Car name to be deleted") @
...
+    @ApiOperation(value = "Repair crashed car", response = Car.class)
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Car successfully repaired"),
+        @ApiResponse(code = 400, message = "Car with given name does not exist")
+    })
+    public Car repairCar(@PathVariable("name") String name){
```
Comment (sebastian-mikucki) [inline src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: should we return whole car after repair?

Comment (sebastian-mikucki) [reply src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: I think http 200 should be enough

File: src/test/java/com/guysfromusa/carsgame/v1/CarApiAware.java [hunk 5]
```diff
@@ -41,4 +41,12 @@ default Long removeCar(TestRestTemplate template, String carName){
         ResponseEntity<Long> modifiedCar =  template.exchange(url, DELETE, null, Long.class);
         return modifiedCar.getBody();
     }
+
+    default Car repairCar(TestRestTemplate template, String carName){
```
Comment (sebastian-mikucki) [inline src/test/java/com/guysfromusa/carsgame/v1/CarApiAware.java]: after introducing @Sql I think we should try to load data directly to db to make tests clean and readable

File: src/main/java/com/guysfromusa/carsgame/services/CarService.java (hunk 1 above)
Comment (tbradlo) [reply src/main/java/com/guysfromusa/carsgame/services/CarService.java]: this means you can skip last line (if you have repo test ;))

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java (hunk 3 above)
Comment (tbradlo) [reply src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: there might be issue with spring test with PUT mapping

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java (hunk 4 above)
Comment (tbradlo) [reply src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: maybe list of running games ;p

File: src/test/java/com/guysfromusa/carsgame/services/CarServiceTest.java [hunk 6]
```diff
@@ -158,4 +167,39 @@ public boolean matches(Object argument) {
...
+    public void shouldNotFindCar(){
+        //given
+        String carName = "Dodge";
+
+        when(carRepository.findByName(carName)).thenReturn(Optional.empty());
+
+        //when
+        expectedException.expect(EntityNotFoundException.class);
+        expectedException.expectMessage("Car with name Dodge does not exist");
+        carService.repairCar(carName);
```
Comment (tbradlo) [inline src/test/java/com/guysfromusa/carsgame/services/CarServiceTest.java]: assertThatExceptionOfType(Abc.).isThrownBy(() -> {
            
        }).withMessage()

File: src/test/java/com/guysfromusa/carsgame/services/CarServiceTest.java [hunk 7]
```diff
@@ -158,4 +167,39 @@ public boolean matches(Object argument) {
...
+    }
+
+    private CarEntity createCarEntity() {
```
Comment (tbradlo) [inline src/test/java/com/guysfromusa/carsgame/services/CarServiceTest.java]: builder

**Pull request closed without merging**